modules/api/src/adapter/buffer.ts

```diff
@@ -0,0 +1,99 @@
+// luma.gl, MIT license
+import type Device from './device';
+import Resource, {ResourceProps, DEFAULT_RESOURCE_PROPS} from './resource';
+
+export type BufferProps = ResourceProps & {
+  handle?: WebGLBuffer;
+
+  byteLength?: number;
+  data?: ArrayBuffer | ArrayBufferView;
+
+  usage?: number;
+  mappedAtCreation?: boolean;
+  target?: number;
+
+  accessor?: any; // AccessorObject;
+
+  /** @deprecated */
+  index?: number;
+  /** @deprecated */
+  offset?: number;
+  /** @deprecated */
+  size?: number;
+  /** @deprecated */
+  type?: number;
+}
+
+// @ts-expect-error
+const DEFAULT_BUFFER_PROPS: Required<BufferProps> = {
+  ...DEFAULT_RESOURCE_PROPS,
+  byteLength: 0,
+  data: undefined,
+  usage: undefined, // GPUBufferUsage.COPY_DST | GPUBufferUsage.COPY_SRC
+  mappedAtCreation: true
+};
+
+/** Abstract GPU buffer */
+export default abstract class Buffer extends Resource<BufferProps> {
+  static MAP_READ = 0x01;
+  static MAP_WRITE = 0x02;
+
+  constructor(device: Device, props: BufferProps) {
+    super(device, props, DEFAULT_BUFFER_PROPS);
+  }
+
+  get [Symbol.toStringTag](): string {
+    return 'Buffer';
```

Comment:
Return id?

Reply:
- [toStringTag](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toStringTag) should just return a constant string. 
- We do have a `toString` in the resource base class that adds the id.